### PR TITLE
Generate parameter documentation for console launcher automatically

### PR DESCRIPTION
**Pull request by signed**
## Overview

Generate parameter documentation for console launcher automatically.
No longer have two places where parameters and descriptions have to be maintained.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: documentation/documentation.gradle [hunk 1]
```diff
@@ -51,6 +51,7 @@ asciidoctorj {
 }
 
 asciidoctor {
+    dependsOn test
```
Comment (sormuras) [inline documentation/documentation.gradle]: Spotless complains about spaces here. Please run "spotlessApply".

File: documentation/src/test/java/documentation/ConsoleLauncherParameterDoc.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import static java.nio.charset.Charset.defaultCharset;
+import static java.nio.charset.Charset.forName;
+
+class ConsoleLauncherParameterDoc {
+
+    private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+    private final PrintStream outSink = new PrintStream(out);
+    private final ByteArrayOutputStream err = new ByteArrayOutputStream();
+    private final PrintStream errSink = new PrintStream(err);
```
Comment (sormuras) [inline documentation/src/test/java/documentation/ConsoleLauncherParameterDoc.java]: Refactor fields to local variables of the `generateHelpOutput()` method below.

**Comment by sbrannen** [comment]
I'm totally fine with having the CLI options automatically generated and imported into the User Guide, but I am not a fan of the "using a test class as a utility" hack.

I would therefore prefer to see this as a custom Gradle task dedicated to that purpose.

**Comment by marcphilipp** [comment]
> I would therefore prefer to see this as a custom Gradle task (potentially even _inlined_ in the build script) dedicated to that purpose.

Agreed and done in 6fb89e9822244d07e5e7b462c787c114fd819e8c.

**Comment by sbrannen** [comment]
Very nice, @marcphilipp! 

Exactly what I had in mind... 👍

**Comment by sbrannen** [comment]
Except... well... it [breaks the build on AppVeyor](https://ci.appveyor.com/project/marcphilipp/junit5/build/1729#L219).

```
:clean
:documentation:clean FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':documentation:clean'.
> Unable to delete file: C:\projects\junit5\documentation\build\generated\asciidoc\console-launcher-options.txt
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

I'm therefore reopening this issue to address the build issue on Windows.

**Comment by marcphilipp** [comment]
Argh... 😞

**Comment by marcphilipp** [comment]
Fixed in 7fc1b67654aba1862e00086a2290c5d79fa0f9c2 and 119600a8a1a28b8e9c341cf91a7923f057218874.

**Pull request closed without merging**